internal/warehouse/repository/mariadb/repository_test.go

```diff
@@ -0,0 +1,348 @@
+
+		db, mock, err := sqlmock.New()
+		assert.NoError(t, err)
+
+		defer db.Close()
+
+		mock.ExpectExec(regexp.QuoteMeta(CreateWarehouse)).WithArgs(
+			mockWarehouse.Address,
+			mockWarehouse.Telephone,
+			mockWarehouse.WarehouseCode,
+			mockWarehouse.MinimunCapacity,
+			mockWarehouse.MinimunTemperature,
+			mockWarehouse.LocalityID,
+		).WillReturnResult(sqlmock.NewResult(1, 1))
+
+		mariadbWarehouse := NewMariadbWarehouseRepository(db)
+
+		newWarehouse, err := mariadbWarehouse.Create(context.TODO(), &mockWarehouse)
+
+		assert.NoError(t, err)
+		assert.Equal(t, expectedWarehouse, newWarehouse)
+	})
+
+	t.Run("error: when ExecContext return an error", func(t *testing.T) {
+		db, mock, err := sqlmock.New()
+		assert.NoError(t, err)
+
+		defer db.Close()
+
+		mock.ExpectExec(regexp.QuoteMeta(CreateWarehouse)).WithArgs(
+			mockWarehouse.Address,
+			mockWarehouse.Telephone,
+			mockWarehouse.WarehouseCode,
+			mockWarehouse.MinimunCapacity,
+			mockWarehouse.MinimunTemperature,
+			mockWarehouse.LocalityID,
+		).WillReturnError(fmt.Errorf("some problem on the query"))
+
+		mariadbWarehouse := NewMariadbWarehouseRepository(db)
+
+		newWarehouse, err := mariadbWarehouse.Create(context.TODO(), &mockWarehouse)
+
+		assert.Error(t, err)
+		assert.Empty(t, newWarehouse)
+	})
+
+}
+
+func Test_repository_update(t *testing.T) {
+	t.Run("update_not_found", func(t *testing.T) {
+
+		db, mock, err := sqlmock.New()
+		assert.NoError(t, err)
+
+		defer db.Close()
+
+		mock.ExpectExec(regexp.QuoteMeta(UpdateWarehouse)).WithArgs(
+			updateWarehouse.MinimunCapacity,
+			updateWarehouse.MinimunTemperature,
+			expectedWarehouse.Id,
+		).WillReturnResult(sqlmock.NewResult(0, 0))
+
+		mariadbWarehouse := NewMariadbWarehouseRepository(db)
+
+		_, err = mariadbWarehouse.Update(context.TODO(), expectedWarehouse.Id, &updateWarehouse)
+
+		assert.Error(t, err)
+
+	})
+
+	t.Run("update_err_exec_query", func(t *testing.T) {
+
+		db, mock, err := sqlmock.New()
+		assert.NoError(t, err)
+
+		defer db.Close()
+
+		errMsg := fmt.Errorf("error: error on query")
+
+		mock.ExpectExec(regexp.QuoteMeta(UpdateWarehouse)).WillReturnError(errMsg)
+
+		mariadbWarehouse := NewMariadbWarehouseRepository(db)
+
+		_, err = mariadbWarehouse.Update(context.TODO(), expectedWarehouse.Id, &updateWarehouse)
+
+		assert.Error(t, err)
+
+	})
+
+	t.Run("update_success", func(t *testing.T) {
+		db, mock, err := sqlmock.New()
+		assert.NoError(t, err)
+		defer db.Close()
+
+		mock.ExpectExec(regexp.QuoteMeta(UpdateWarehouse)).WithArgs(
+			updateWarehouse.MinimunCapacity,
+			updateWarehouse.MinimunTemperature,
+			expectedWarehouse.Id,
+		).WillReturnResult(sqlmock.NewResult(0, 1))
+
+		mariadbWarehouse := NewMariadbWarehouseRepository(db)
+
+		expectReturn, err := mariadbWarehouse.Update(context.Background(), expectedWarehouse.Id, &updateWarehouse)
+
+		assert.NoError(t, err)
+		assert.Equal(t, expectReturn, updateWarehouse)
+	})
+}
```

Comment:
Sugestão, fazer um teste para "update_not_ok: return error when rows not affected" 
este teste vai pegar aquele if na linha 136 do Update